Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -147,7 +147,7 @@ public struct NIOAsyncWriter<
     /// This struct contains two properties:
     /// 1. The ``sink`` which should be retained by the consumer and is used to set the writability.
     /// 2. The ``writer`` which is the actual ``NIOAsyncWriter`` and should be passed to the producer.
-    public struct NewWriter {
+    public struct NewWriter: @unchecked Sendable {
```

Comment:
Unsure about that one the `Sink` is not `Sendable` right now because we expect it to be only held in one isolation domain at any given time even though it is fully thread-safe. We certainly can't make this type `unchecked Sendable` then. So we either make the `Sink` `Sendable` as well or don't make this one `Sendable`